### PR TITLE
LPS-178992 | Poshi Automation for Bug | Master 

### PR DESCRIPTION
**Pull request by rafaellajordao**
# LPS-178992 - Poshi Automation for Bug

Hi! :)
This pr is corresponding to this ticket: [LPS-138908](https://issues.liferay.com/browse/LPS-138908) → [LPS-178992](https://issues.liferay.com/browse/LPS-178992)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 1]
```diff
@@ -741,6 +741,47 @@ definition {
 		}
 	}
 
+	@description = "This is a test for Bug LPS-138908. Type filter in Content Dashboard does work when deselecting previously selected types"
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:744]: This is a test for LPS-138908.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 2]
```diff
@@ -741,6 +741,47 @@ definition {
...
+	@priority = 4
+	test CanDeselectTypesInTypeFilter {
+		task ("Given the content dashboard page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:747]: Given a user has a Content Dashboard filtered by a type

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 3]
```diff
@@ -741,6 +741,47 @@ definition {
...
+
+			ContentDashboard.goToContentDashboard();
+		}
+
+		task ("When deselect previously selected types") {
+			ContentDashboard.filterByType(typeName = "Web Content Article");
+
+			AssertTextEquals(
+				locator1 = "ContentDashboard#FIRST_ELEMENT_LISTED",
+				value1 = "WC Content");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:767]: LexiconTable.viewTableEntryTitle(tableEntry = "WC Content");

LexiconTable.viewTableEntryNotPresent(tableEntry = "tree.png");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 4]
```diff
@@ -741,6 +741,47 @@ definition {
...
+			JSONDocument.addFileWithUploadedFile(
+				dmDocumentTitle = "Document_1",
+				groupName = "Test Site Name",
+				mimeType = "application/msword",
+				sourceFileName = "Document_1.doc");
+
+			ContentDashboard.goToContentDashboard();
+		}
+
+		task ("When deselect previously selected types") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:762]: Merge into Given task block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 5]
```diff
@@ -741,6 +741,47 @@ definition {
...
+			ContentDashboard.filterByType(typeName = "Web Content Article");
+
+			AssertTextEquals(
+				locator1 = "ContentDashboard#FIRST_ELEMENT_LISTED",
+				value1 = "WC Content");
+		}
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:769]: When the user deselects the type filter

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 6]
```diff
@@ -741,6 +741,47 @@ definition {
...
+			JSONWebcontent.addWebContent(
+				content = "Webcontent Content",
+				groupName = "Test Site Name",
+				title = "WC Content");
+
+			JSONDocument.addFileWithUploadedFile(
+				dmDocumentTitle = "Document_1",
+				groupName = "Test Site Name",
+				mimeType = "application/msword",
+				sourceFileName = "Document_1.doc");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:757]: No need. Use ootb document, like tree.png

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 7]
```diff
@@ -741,6 +741,47 @@ definition {
...
+
+		task ("Then apply the filter with no types or subtypes selected and all content would be visible then") {
+			ContentDashboard.openFilterModal(filterOption = "Type");
+
+			Uncheck.uncheckNotVisible(
+				key_nodeName = "Web Content Article",
+				locator1 = "Treeview#NODE_CHECKBOX_2");
+
+			ContentDashboard.applySelectedFilter();
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:778]: Then the user sees all assets

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 8]
```diff
@@ -741,6 +741,47 @@ definition {
...
+			AssertTextEquals(
+				locator1 = "ContentDashboard#FIRST_ELEMENT_LISTED",
+				value1 = "Document_1");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:781]: LexiconTable.viewTableEntryTitle(tableEntry = "WC Content");

LexiconTable.viewTableEntryTitle(tableEntry = "tree.png");

**Review by Tim-Cao — COMMENTED** [review]
Hi @rafaellajordao, would you mind refactoring according to comments? Thanks

**Review by Tim-Cao — COMMENTED** [review]
Hi @rafaellajordao, would you mind refactoring according to comments? Thanks

**Review by Tim-Cao — COMMENTED** [review]
Hi @rafaellajordao, would you mind refactoring according to comments? Thanks

**Review by Tim-Cao — COMMENTED** [review]
Hi @rafaellajordao, would you mind refactoring according to comments? Thanks

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
LGTM!

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**